source/modulo_components/modulo_components/component_interface.py

```diff
@@ -67,29 +67,159 @@ def __step(self):
+        except (AttributeError, TypeError) as e:
+            self.get_logger().error(f"Failed to add parameter: {e}")
+            return
+        ros_param = write_parameter(sr_parameter)
+        if not self.has_parameter(sr_parameter.get_name()):
+            self._parameter_dict[sr_parameter.get_name()] = parameter
+            # TODO ignore override
+            self.declare_parameter(ros_param.name, ros_param.value,
+                                   descriptor=ParameterDescriptor(description=description), ignore_override=read_only)
+        else:
+            self.set_parameters([ros_param])
+
+    def get_parameter(self, name: str) -> Union[sr.Parameter, rclpy.parameter.Parameter]:
+        """
+        Get a parameter by name. If this method is called from a file that contains 'rclpy' in its path, the
+        rclpy.node.Node.get_parameter method will be invoked, otherwise return the parameter from the local parameter
+        dictionary.
+
+        :param name: The name of the parameter
+        :return: The requested parameter
+        """
+        co_filename = sys._getframe().f_back.f_code.co_filename
+        self.get_logger().debug(f"get_parameter called from {co_filename}")
+        if "rclpy" in co_filename:
+            return rclpy.node.Node.get_parameter(self, name)
+        else:
+            return self._get_component_parameter(name)
+
+    def _get_component_parameter(self, name: str) -> sr.Parameter:
+        """
+        Get the parameter from the parameter dictionary by its name.
+
+        :param name: The name of the parameter
+        :return: The parameter, if it exists
+        :raises ComponentParameterError if the parameter does not exist
+        """
+        if name not in self._parameter_dict.keys():
+            raise ComponentParameterError(f"Failed to get parameter '{name}'")
+        if isinstance(self._parameter_dict[name], str):
+            return self.__getattribute__(self._parameter_dict[name])
+        else:
+            return self._parameter_dict[name]
+
+    def get_parameter_value(self, name: str) -> T:
+        """
+        Get the parameter value from the parameter dictionary by its name.
+
+        :param name: The name of the parameter
+        :return: The value of the parameter, if the parameter exists
+        :raises ComponentParameterError if the parameter does not exist
+        """
+        return self._get_component_parameter(name).get_value()
+
+    def set_parameter_value(self, name: str, value: T, parameter_type: sr.ParameterType) -> None:
+        """
+        Set the value of a parameter. The parameter must have been previously declared. If the parameter is an
+        attribute, the attribute is updated.
+
+        :param name: The name of the parameter
+        :param value: The value of the parameter
+        :param parameter_type: The type of the parameter
+        """
+        result = self.set_parameters([write_parameter(sr.Parameter(name, value, parameter_type))])[0]
```

Comment:
I think this would be more readable in multiple lines.

In addition, we could add an explicit check for if parameter exists before we go through the effort of trying to set it.

```python
if name not in self._parameter_dict.keys():
    raise ComponentParameterError(f"Failed to get parameter '{name}'")

ros_param = write_parameter(sr.Parameter(name, value, parameter_type))
result = self.set_parameters([ros_param])[0]
```